packages/dds/tree/src/test/util/summaryBenchmarkSizes.ts

```diff
@@ -0,0 +1,142 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { makeRandom } from "@fluid-internal/stochastic-test-utils";
+import { IsoBuffer } from "@fluidframework/common-utils";
+import { TransactionResult } from "../../checkout";
+import { singleTextCursor } from "../../feature-libraries";
+import { ISharedTree } from "../../shared-tree";
+import { detachedFieldAsKey, TreeValue } from "../../tree";
+import { brand } from "../../util";
+import { TestTreeProvider } from "../utils";
+
+const random = makeRandom(129384);
+
+export interface OutputType {
+    noEdits: number,
+    insert1: number,
+    insert10: number,
+    insert100: number,
+    insertRemove1: number,
+    insertRemove10: number,
+    insertRemove100: number,
+}
+
+export async function getSummaryBenchmarkSizes(): Promise<OutputType> {
+
+    const output: OutputType = {
+        noEdits: 0,
+        insert1: 0,
+        insert10: 0,
+        insert100: 0,
+        insertRemove1: 0,
+        insertRemove10: 0,
+        insertRemove100: 0,
+    }
+
+    // summary size for tree with no edit.
+    const provider = await TestTreeProvider.create(1);
+    const tree = provider.trees[0];
+
+    const summarize = await provider.enableManualSummarization();
+    await summarize();
+    const summary = JSON.stringify(tree.getAttachSummary());
+    output.noEdits = IsoBuffer.from(summary).byteLength;
+
+    // summary sizes for inserting nodes.
+    output.insert1 = await getInsertsSummarySize(1);
+    output.insert10 = await getInsertsSummarySize(10);
+    output.insert100 = await getInsertsSummarySize(100);
+
+    // summary sizes for inserting then immediately removing nodes.
+    output.insertRemove1 = await getInsertsSummarySize(1, true);
+    output.insertRemove10 = await getInsertsSummarySize(10, true);
+    output.insertRemove100 = await getInsertsSummarySize(100, true);
```

Comment:
shared-tree summaries consist mainly of a collection summaries of Indexes. Currently this is just the  SchemaIndex and ForestIndex. Neither of these include anything from the op history, so a bunch of inserts and deletes doesn't make a lot of sense. I'd expect to be testing ForestIndex with various tree shapes and sizes and SchemaIndex with various schema.

While it might be useful to do bunch of inserts then removes and assert the summary is not impacted by that, that seems like a job for a shared-tree unit test, and doesn't require all these different variants and stats collection.

Reply:
I have removed these cases for now, and replaced them with a different set of inserts (but with the nodes inserted depth-wise rather than width-wise)